src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -818,6 +818,35 @@ void QgsLayoutItemLegend::onAtlasEnded()
   updateFilterByMap();
 }
 
+QgsExpressionContext QgsLayoutItemLegend::createExpressionContext() const
+{
+  QgsExpressionContext context = QgsLayoutItem::createExpressionContext();
+
+  //Can't utilize QgsExpressionContextUtils::mapSettingsScope as we don't always
+  //have a QgsMapSettings object available when the context is required, so we manually
+  //add the same variables here
+
+  // add the existing context of the linked map
+  if (mMap)
+  {
+    context.appendScope( mMap.createExpressionContext() );
```

Comment:
I think I'll need to tweak the QgsLayoutItemMap API to allow this -- here we want to append just the map specific scope, instead of all the scopes within the map's expression context (since that will contain duplicate general/project/layout/atlas scopes). I'll push a PR which exposes a method to get just the map scope for you to use here.

Reply:
Yeah I wasn't sure if that was useful or if it would simply take more memory but I still included it just to get your opinion on the usefulness of it.

Reply:
It's the right move -- it will be very handy here, because it will give access to variables like the map scale and extent.